File: PrimeChecker.java
/**
 * This class provides a method to check whether a given number is prime.
 */
public class PrimeChecker {
    /**
     * Checks if the given number is prime.
     * @param num The number to check.
     * @return true if the number is prime, false otherwise.
     */
    public static boolean isPrime(int num) {
        // Numbers less than or equal to 1 are not prime
        if (num <= 1) {
            return false;
        }
        // Check for factors up to the square root of the number
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                // If a factor is found, the number is not prime
                return false;
            }
        }
        // If no factors are found, the number is prime
        return true;
    }
}

File: C/prime_checker.c
#include <stdio.h>

int isPrime(int n) {
    if (n <= 1) {
        return 0;  // Not prime
    }
    for (int i = 2; i * i <= n; i++) {
        if (n % i == 0) {
            return 0;  // Not prime
        }
    }
    return 1;  // Prime
}

int main() {
    int num;
    printf("Enter a number to check if it's prime: ");
    scanf("%d", &num);
    if (isPrime(num)) {
        printf("%d is a prime number\n", num);
    } else {
        printf("%d is not a prime number\n", num);
    }
    return 0;
}
